UnitTestFrameworkPkg/GoogleTest.md

```diff
@@ -0,0 +1,1106 @@
+# GoogleTest - Unit Testing Framework
+
+## About
+This unit test framework, called **GoogleTest** is implemented as a set of EDK II libraries. It is one of two unit test frameworks supported by EDK II. Please refer to [the ReadMe](./ReadMe.md) for a comparison of the two.
+[GoogleTest](http://google.github.io/googletest/) and can be used to implement host-based unit tests. [GoogleTest on GitHub](https://github.com/google/googletest) is included in the UnitTestFrameworkPkg as a submodule. Use of GoogleTest for target-based unit tests of EDK II components is not supported.
+Host-based unit tests that require mocked interfaces can use the mocking infrastructure included with GoogleTest called [gMock](https://github.com/google/googletest/tree/main/googlemock).
+
+
+## GoogleTest Samples
+
+There is a sample unit test provided as both an example of how to write a unit test and leverage
+many of the GoogleTest features. This sample can be found in the `Test/GoogleTest/Sample/SampleGoogleTest`
+directory.
+
+The sample is provided for the HOST_APPLICATION build type, which can be run on a host system without
+needing a target.
+
+There is also a sample unit test provided as both an example of how to write a unit test with
+mock functions and leverage some of the gMock features. This sample can be found in the
+`SecurityPkg/Library/SecureBootVariableLib/GoogleTest` directory.
+
+It too is provided for the HOST_APPLICATION build type, which can be run on a host system without
+needing a target.
+
+## GoogleTest Usage
+
+This section is built a lot like a "Getting Started". We'll go through some of the components that are needed
+when constructing a unit test and some of the decisions that are made by the test writer. We'll also describe
+how to check for expected conditions in test cases and a bit of the logging characteristics.
+
+Most of these examples will refer to the `SampleGoogleTestHost` app found in this package, but
+the examples related to mock functions will refer to the `SecureBootVariableLibGoogleTest` app
+found in the `SecurityPkg`.
+
+There's an instructional video that walks through the creation of a new GoogleTest in UEFI:
+[![Watch the Video Here](https://img.youtube.com/vi/Ufc3P95MqJE/default.jpg)](https://www.youtube.com/watch?v=Ufc3P95MqJE)
+
+### GoogleTest Requirements - INF
+
+In our INF file, we'll need to bring in the `GoogleTestLib` library. Conveniently, the interface
+header for the `GoogleTestLib` is in `UnitTestFrameworkPkg`, so you shouldn't need to depend on any other
+packages. As long as your DSC file knows where to find the lib implementation that you want to use,
+you should be good to go.
+
+See this example in `SampleGoogleTestHost.inf`...
+
+```inf
+[Packages]
+  MdePkg/MdePkg.dec
+  UnitTestFrameworkPkg/UnitTestFrameworkPkg.dec
+
+[LibraryClasses]
+  GoogleTestLib
+  BaseLib
+  DebugLib
+```
+
+Also, if you want your test to automatically be picked up by the Test Runner plugin, you will need

```

Comment:
Please explain or link to what "Test Runner" plugin means here. I imagine most people reading wouldn't know why they would want to do that.